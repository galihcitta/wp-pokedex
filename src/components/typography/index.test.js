import React from "react";
import { HeadingText, Text } from "./index";

describe("Header Menu Lang Component", () => {
    it("renders and create snapshot <Text.BodyOne />", () => {
        expect(
            toJson(mount(<Text.BodyOne fontWeight="bold"></Text.BodyOne>)),
        ).toMatchSnapshot();
    });

    it("renders and check fontWeight Bold <Text.Caption />", () => {
        expect(
            mount(<Text.Caption fontWeight="bold"></Text.Caption>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <Text.BodyTwo />", () => {
        expect(
            toJson(mount(<Text.Caption fontWeight="bold"></Text.Caption>)),
        ).toMatchSnapshot();
    });

    it("renders and check fontWeight Bold <Text.BodyTwo />", () => {
        expect(
            mount(<Text.Caption fontWeight="bold"></Text.Caption>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and check fontWeight Bold <Text.BodyOne />", () => {
        expect(
            mount(<Text.BodyOne fontWeight="bold"></Text.BodyOne>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <Text.BodyTwo />", () => {
        expect(
            toJson(mount(<Text.BodyOne fontWeight="bold"></Text.BodyOne>)),
        ).toMatchSnapshot();
    });

    it("renders and check fontWeight Bold <Text.BodyTwo />", () => {
        expect(
            mount(<Text.BodyOne fontWeight="bold"></Text.BodyOne>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });

    it("renders and check fontWeight Bold <Text.BodyOne />", () => {
        expect(
            mount(<Text.BodyTwo fontWeight="bold"></Text.BodyTwo>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <Text.BodyTwo />", () => {
        expect(
            toJson(mount(<Text.BodyTwo fontWeight="bold"></Text.BodyTwo>)),
        ).toMatchSnapshot();
    });

    it("renders and check fontWeight Bold <Text.BodyTwo />", () => {
        expect(
            mount(<Text.BodyTwo fontWeight="bold"></Text.BodyTwo>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });

    it("renders and check fontWeight Bold <Text.SubtitleOne />", () => {
        expect(
            mount(<Text.SubtitleOne fontWeight="bold"></Text.SubtitleOne>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <Text.SubtitleOne />", () => {
        expect(
            toJson(
                mount(<Text.SubtitleOne fontWeight="bold"></Text.SubtitleOne>),
            ),
        ).toMatchSnapshot();
    });

    it("renders and check block <Text.SubtitleTwo />", () => {
        expect(
            mount(<Text.SubtitleOne block></Text.SubtitleOne>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check fontWeight Bold <Text.SubtitleTwo />", () => {
        expect(
            mount(<Text.SubtitleTwo fontWeight="bold"></Text.SubtitleTwo>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <Text.SubtitleTwo />", () => {
        expect(
            toJson(
                mount(<Text.SubtitleTwo fontWeight="bold"></Text.SubtitleTwo>),
            ),
        ).toMatchSnapshot();
    });

    it("renders and check fontWeight Bold <Text.SubtitleTwo />", () => {
        expect(
            mount(<Text.Overline fontWeight="bold"></Text.Overline>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <Text.Overline />", () => {
        expect(
            toJson(mount(<Text.Overline fontWeight="bold"></Text.Overline>)),
        ).toMatchSnapshot();
    });

    it("renders and check block <Text.Overline />", () => {
        expect(
            mount(<Text.Overline block></Text.Overline>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check block <Text.Caption />", () => {
        expect(
            mount(<Text.Caption block></Text.Caption>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check fontWeight Bold <Text.Caption />", () => {
        expect(
            mount(<Text.Caption fontWeight="bold"></Text.Caption>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <Text.Caption />", () => {
        expect(
            toJson(mount(<Text.Caption fontWeight="bold"></Text.Caption>)),
        ).toMatchSnapshot();
    });

    it("renders and check block <Text.Caption />", () => {
        expect(
            mount(<Text.Caption block></Text.Caption>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check block <HeadingText.One />", () => {
        expect(
            mount(<HeadingText.One block></HeadingText.One>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check fontWeight Bold <HeadingText.One />", () => {
        expect(
            mount(<HeadingText.One fontWeight="bold"></HeadingText.One>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <HeadingText.One />", () => {
        expect(
            toJson(
                mount(<HeadingText.One fontWeight="bold"></HeadingText.One>),
            ),
        ).toMatchSnapshot();
    });

    it("renders and check block <HeadingText.One />", () => {
        expect(
            mount(<HeadingText.One block></HeadingText.One>).prop("block"),
        ).toEqual(true);
    });
    it("renders and check block <HeadingText.Two />", () => {
        expect(
            mount(<HeadingText.Two block></HeadingText.Two>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check fontWeight Bold <HeadingText.Two />", () => {
        expect(
            mount(<HeadingText.Two fontWeight="bold"></HeadingText.Two>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <HeadingText.Two />", () => {
        expect(
            toJson(
                mount(<HeadingText.Two fontWeight="bold"></HeadingText.Two>),
            ),
        ).toMatchSnapshot();
    });

    it("renders and check fontWeight Bold <HeadingText.Three />", () => {
        expect(
            mount(
                <HeadingText.Three fontWeight="bold"></HeadingText.Three>,
            ).prop("fontWeight"),
        ).toEqual("bold");
    });
    it("renders and create snapshot <HeadingText.Three />", () => {
        expect(
            toJson(
                mount(
                    <HeadingText.Three fontWeight="bold"></HeadingText.Three>,
                ),
            ),
        ).toMatchSnapshot();
    });

    it("renders and check block <HeadingText.Three />", () => {
        expect(
            mount(<HeadingText.Three block></HeadingText.Three>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check fontWeight Bold <HeadingText.Four />", () => {
        expect(
            mount(<HeadingText.Four fontWeight="bold"></HeadingText.Four>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <HeadingText.Four />", () => {
        expect(
            toJson(
                mount(<HeadingText.Four fontWeight="bold"></HeadingText.Four>),
            ),
        ).toMatchSnapshot();
    });

    it("renders and check block <HeadingText.Four />", () => {
        expect(
            mount(<HeadingText.Four block></HeadingText.Four>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check fontWeight Bold <HeadingText.Five />", () => {
        expect(
            mount(<HeadingText.Five fontWeight="bold"></HeadingText.Five>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <HeadingText.Five />", () => {
        expect(
            toJson(
                mount(<HeadingText.Five fontWeight="bold"></HeadingText.Five>),
            ),
        ).toMatchSnapshot();
    });

    it("renders and check block <HeadingText.Five />", () => {
        expect(
            mount(<HeadingText.Five block></HeadingText.Five>).prop("block"),
        ).toEqual(true);
    });

    it("renders and check fontWeight Bold <HeadingText.Six />", () => {
        expect(
            mount(<HeadingText.Six fontWeight="bold"></HeadingText.Six>).prop(
                "fontWeight",
            ),
        ).toEqual("bold");
    });
    it("renders and create snapshot <HeadingText.Six />", () => {
        expect(
            toJson(
                mount(<HeadingText.Six fontWeight="bold"></HeadingText.Six>),
            ),
        ).toMatchSnapshot();
    });

    it("renders and check block <HeadingText.Six />", () => {
        expect(
            mount(<HeadingText.Six block></HeadingText.Six>).prop("block"),
        ).toEqual(true);
    });
});
