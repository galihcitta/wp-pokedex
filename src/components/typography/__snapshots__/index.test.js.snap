// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Menu Lang Component renders and create snapshot <HeadingText.Five /> 1`] = `
.c0 {
  font-family: "Muli",sans-serif;
  font-size: 24.4px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: normal;
  text-align: left;
}

<styled.h5
  fontWeight="bold"
>
  <h5
    className="c0"
    fontWeight="bold"
  />
</styled.h5>
`;

exports[`Header Menu Lang Component renders and create snapshot <HeadingText.Four /> 1`] = `
.c0 {
  font-family: "Rubik",sans-serif;
  font-size: 34.5px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  line-height: normal;
  text-align: left;
}

<styled.h4
  fontWeight="bold"
>
  <h4
    className="c0"
    fontWeight="bold"
  />
</styled.h4>
`;

exports[`Header Menu Lang Component renders and create snapshot <HeadingText.One /> 1`] = `
.c0 {
  font-family: "Rubik",sans-serif;
  font-size: 60px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: normal;
}

<styled.h1
  fontWeight="bold"
>
  <h1
    className="c0"
    fontWeight="bold"
  />
</styled.h1>
`;

exports[`Header Menu Lang Component renders and create snapshot <HeadingText.Six /> 1`] = `
.c0 {
  font-family: "Muli",sans-serif;
  font-size: 20.3px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  line-height: normal;
  text-align: left;
}

<styled.h6
  fontWeight="bold"
>
  <h6
    className="c0"
    fontWeight="bold"
  />
</styled.h6>
`;

exports[`Header Menu Lang Component renders and create snapshot <HeadingText.Three /> 1`] = `
.c0 {
  font-family: "Rubik",sans-serif;
  font-size: 48.8px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: normal;
  text-align: left;
}

<styled.h3
  fontWeight="bold"
>
  <h3
    className="c0"
    fontWeight="bold"
  />
</styled.h3>
`;

exports[`Header Menu Lang Component renders and create snapshot <HeadingText.Two /> 1`] = `
.c0 {
  font-family: "Rubik",sans-serif;
  font-size: 45px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: normal;
  text-align: left;
}

<styled.h2
  fontWeight="bold"
>
  <h2
    className="c0"
    fontWeight="bold"
  />
</styled.h2>
`;

exports[`Header Menu Lang Component renders and create snapshot <Text.BodyOne /> 1`] = `
.c0 {
  font-family: "Muli",sans-serif;
  font-size: 16.3px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.72;
  text-align: left;
}

<styled.p
  fontWeight="bold"
>
  <p
    className="c0"
    fontWeight="bold"
  />
</styled.p>
`;

exports[`Header Menu Lang Component renders and create snapshot <Text.BodyTwo /> 1`] = `
.c0 {
  display: inline-block;
  font-family: "Muli",sans-serif;
  font-size: 12.2px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  line-height: 1.31;
  text-align: left;
}

<styled.span
  fontWeight="bold"
>
  <span
    className="c0"
    fontWeight="bold"
  />
</styled.span>
`;

exports[`Header Menu Lang Component renders and create snapshot <Text.BodyTwo /> 2`] = `
.c0 {
  font-family: "Muli",sans-serif;
  font-size: 16.3px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.72;
  text-align: left;
}

<styled.p
  fontWeight="bold"
>
  <p
    className="c0"
    fontWeight="bold"
  />
</styled.p>
`;

exports[`Header Menu Lang Component renders and create snapshot <Text.BodyTwo /> 3`] = `
.c0 {
  font-family: "Muli",sans-serif;
  font-size: 14.2px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  line-height: 1.41;
  text-align: left;
}

<styled.p
  fontWeight="bold"
>
  <p
    className="c0"
    fontWeight="bold"
  />
</styled.p>
`;

exports[`Header Menu Lang Component renders and create snapshot <Text.Caption /> 1`] = `
.c0 {
  display: inline-block;
  font-family: "Muli",sans-serif;
  font-size: 12.2px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  line-height: 1.31;
  text-align: left;
}

<styled.span
  fontWeight="bold"
>
  <span
    className="c0"
    fontWeight="bold"
  />
</styled.span>
`;

exports[`Header Menu Lang Component renders and create snapshot <Text.Overline /> 1`] = `
.c0 {
  font-family: "Muli",sans-serif;
  font-size: 12.2px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  line-height: 1.31;
  text-align: left;
}

<styled.h5
  fontWeight="bold"
>
  <h5
    className="c0"
    fontWeight="bold"
  />
</styled.h5>
`;

exports[`Header Menu Lang Component renders and create snapshot <Text.SubtitleOne /> 1`] = `
.c0 {
  display: inline-block;
  font-family: "Muli",sans-serif;
  font-size: 16.3px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
  line-height: 1.48;
  text-align: left;
}

<styled.span
  fontWeight="bold"
>
  <span
    className="c0"
    fontWeight="bold"
  />
</styled.span>
`;

exports[`Header Menu Lang Component renders and create snapshot <Text.SubtitleTwo /> 1`] = `
.c0 {
  display: inline-block;
  font-family: "Muli",sans-serif;
  font-size: 14.2px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  line-height: 1.69;
  text-align: left;
}

<styled.span
  fontWeight="bold"
>
  <span
    className="c0"
    fontWeight="bold"
  />
</styled.span>
`;
